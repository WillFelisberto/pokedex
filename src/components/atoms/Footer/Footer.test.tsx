import { render, screen } from '@/tests/test-utils';

import { Footer } from '.';

describe('<Footer />', () => {
  it('should render the footer with correct text and link', () => {
    const { container } = render(<Footer />);

    expect(screen.getByText(/Made with/i)).toBeInTheDocument();
    expect(screen.getByText(/Willian Felisberto/i)).toBeInTheDocument();

    const link = screen.getByRole('link', { name: /Willian Felisberto/i });
    expect(link).toHaveAttribute('href', 'https://github.com/willFelisberto/');

    expect(container).toMatchSnapshot();
  });

  it('should be positioned at the bottom of the screen', () => {
    render(<Footer />);

    const footer = screen.getByRole('contentinfo');
    expect(footer).toHaveClass('bg-transparent', 'bottom-0', 'w-full');
  });
});
