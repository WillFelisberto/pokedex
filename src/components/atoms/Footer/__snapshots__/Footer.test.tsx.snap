// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render the footer with correct text and link 1`] = `
<div>
  <footer
    class="bg-gray-200 bottom-0 w-full text-gray-500 text-center p-4"
  >
    Made with 
    <span
      class="text-red-500"
    >
      ❤️
    </span>
     by
     
    <a
      class="underline  text-gray-500 hover:text-gray-800"
      href="https://github.com/willFelisberto/"
      rel="noopener noreferrer"
      target="_blank"
    >
      Willian Felisberto
    </a>
  </footer>
</div>
`;
